packages/vite-plugin-cloudflare/src/miniflare-options.ts

```diff
@@ -299,77 +309,98 @@ export function getDevMiniflareOptions(
 
 	const workersFromConfig =
 		resolvedPluginConfig.type === "workers"
-			? Object.entries(resolvedPluginConfig.workers).map(
-					([environmentName, workerConfig]) => {
-						const miniflareWorkerOptions = unstable_getMiniflareWorkerOptions(
-							{
-								...workerConfig,
-								assets: undefined,
-							},
-							resolvedPluginConfig.cloudflareEnv
-						);
-
-						const { externalWorkers } = miniflareWorkerOptions;
-
-						const { ratelimits, ...workerOptions } =
-							miniflareWorkerOptions.workerOptions;
-
-						return {
-							externalWorkers,
-							worker: {
-								...workerOptions,
-								name: workerOptions.name ?? workerConfig.name,
-								unsafeInspectorProxy: inspectorPort !== false,
-								modulesRoot: miniflareModulesRoot,
-								unsafeEvalBinding: "__VITE_UNSAFE_EVAL__",
-								serviceBindings: {
-									...workerOptions.serviceBindings,
-									...(environmentName ===
-										resolvedPluginConfig.entryWorkerEnvironmentName &&
-									workerConfig.assets?.binding
-										? {
-												[workerConfig.assets.binding]: ASSET_WORKER_NAME,
-											}
-										: {}),
-									__VITE_INVOKE_MODULE__: async (request) => {
-										const payload =
-											(await request.json()) as vite.CustomPayload;
-										const invokePayloadData = payload.data as {
-											id: string;
-											name: string;
-											data: [string, string, FetchFunctionOptions];
-										};
-
-										assert(
-											invokePayloadData.name === "fetchModule",
-											`Invalid invoke event: ${invokePayloadData.name}`
+			? await Promise.all(
+					Object.entries(resolvedPluginConfig.workers).map(
+						async ([environmentName, workerConfig]) => {
+							let mixedModeSession: Experimental_MixedModeSession | undefined;
```

Comment:
Does this work? Does this not recreate the mixed mode session every time `getDevMiniflareOptions` is called?

I think I'd expect the mixed mode session to be cached in a Map or something, and `updateBindings()` called